### ISSUE TITLE
Auxiliary loss: add an L2 penalty to the subcomponent factorized parameters

### ISSUE THREAD

**Issue by danbraunai** [goodfire-ai/spd#7]
**Migrated from:** goodfire-ai/spd-gf#57
**Original author:** @leesharkey

---

Auxiliary loss: add an L2 penalty to the subcomponent factorized parameters

Without an L2 loss (or a schatten penalty) there is only an _indirect_ reason for a decomposition to get rid of components that don't add any functionality to the set of components (e.g. components that don't activate). The reason we can expect them to disappear is if we assume that newtorks are underparameterized so that their parameters all kind of do something. Since they all do something, the network needs to learn components that both sum to the parameters of the original network AND help with recon loss when active. 

But if they don't ever do anything (i.e. they don't help with recon loss because e.g. they're never active) then we don't have a huge incentive to get rid of them. 

It might help speed up the process of getting rid of useless components to shrink them all constantly by a small amount. 

This issue should also involve evaluating whether or not this aux loss actually helps training, i.e. if it makes the range of hyperparameters for which SPD works wider, or reaches a solution faster, or reduces seed dependence.

**Comment by danbraunai** [comment]
Worth noting that @Laplace418 has played around with adding weight decay to Adam, and it didn't seem to make a difference. We may want to simply do this. We'll wait until we have a more extensive set of evaluations/toy models to test this on.

**Comment by danbraunai-goodfire** [comment]
No longer an issue. Possible fixed by our delta component but unsure.

**Comment by ocg-goodfire** [comment]
Closing per Oli (triage 2026-07-01): no longer an issue — likely resolved by the delta component. Weight decay experiments earlier showed no effect.
